docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+    // Insert the component into the page
+    self.apos.template.prepend({
+      where: 'head',  // or 'body', 'footer', etc.
+      when: 'hmr',    // Show only when HMR is active
+      bundler: 'vite', // Ensure Vite is the active bundler
+      component: 'my-module:devTools'
+    });
+  }
+};
+```
+  <template v-slot:caption>
+    modules/devTools/index.js
+  </template>
+
+</AposCodeBlock>
+
+### Visibility Options
+
+The `when` option controls when your component appears:
+
+```javascript
+when: 'hmr'   // Only visible when HMR is active
+when: 'dev'   // Visible in any development mode
+when: 'prod'  // Only visible in production
+```
+
+You can combine these with `bundler: 'vite'` to ensure your component only appears when using the Vite bundler.
+
+### Common Use Cases
+
+- Development toolbars
+- Debug information
+- Performance monitoring
+- Asset reload buttons
+
+::: warning
+Remember that components marked with `when: 'dev'` or `when: 'hmr'` will never appear in production, regardless of other settings.
+:::
+
+## Migration and Technical Considerations
+
+### Moving from Webpack to Vite
+
+When migrating your ApostropheCMS project to use Vite, you'll need to make a few code adjustments:
+
+1. **Update CSS/Sass Imports**
+   ```scss
+   // Old webpack style
+   @import "~normalize.css";
+   
+   // New Vite style - remove the ~ prefix
+   @import "normalize.css";
+   ```
+
+2. **Update Module Import Paths**
+   ```javascript
+   // Not recommended
+   import Component from 'apostrophe/modules/module-name/ui/apos/components/Component.vue';
+   
+   // Recommended - use the alias
+   import Component from 'Modules/module-name/components/Component.vue';
+   ```
+
+3. **Convert to ESM Syntax**
+   ```javascript
+   // Remove CommonJS syntax
+   const myComponent = require('./component');
+   module.exports = myComponent;
+
+   // Use ESM instead
+   import myComponent from './component';
+   export default myComponent;
+   ```
+
+### Known Limitations and Solutions
+
+### HMR Directory Watching
+- **Limitation**: HMR only watches existing `anyModule/ui` directories
+- **Solution**: After adding new `ui` directories, restart your development server:
+  ```bash
+  # If using nodemon (default)
+  # Type 'rs' and press Enter in your terminal
+  rs
+  ```
+
+### Static Assets in `ui/public`
+- **Limitation**: Changes to `ui/public` don't trigger HMR/page reload
+- **Workaround**: Add these directories to your `nodemon` watch list:
+  ```json
+  {
+    "watch": [
+      "app.js",
+      "./modules/**/*",
+      "./lib/**/*",
+      "./src/**/*",
+      "**/ui/public/**/*"
+    ]
+  }
+  ```
+
+### Troubleshooting Common Issues
+
+### Missing Hot Updates
+If changes aren't reflecting immediately:
+1. Check your browser console for HMR connection errors
+2. Verify the module's `ui` directory is being watched
+3. Ensure you're using ESM syntax in affected files
+
+### Build Errors
+```bash
+# Clear your build cache
+rm -rf apos-build
```

Comment:
`node app @apostrophecms/asset:reset` is a better solution

Reply:
Replaced - I still have to document this. It essentially does the same thing as rm -rf without have permission warnings or fat finger mistake deletions?